rogue-thi-app/components/cards/RoomCard.js

```diff
@@ -0,0 +1,97 @@
+import React, { useEffect, useState } from 'react'
+import ListGroup from 'react-bootstrap/ListGroup'
+import { useRouter } from 'next/router'
+
+import { faDoorOpen } from '@fortawesome/free-solid-svg-icons'
+
+import BaseCard from './BaseCard'
+
+import { BUILDINGS_ALL, filterRooms, getNextValidDate } from '../../lib/backend-utils/rooms-utils'
+import { formatFriendlyTime, formatISODate, formatISOTime } from '../../lib/date-utils'
+import { NoSessionError } from '../../lib/backend/thi-session-handler'
+import ReactPlaceholder from 'react-placeholder/lib'
+import { getFriendlyTimetable } from '../../lib/backend-utils/timetable-utils'
+/**
+ * Dashboard card for semester and exam dates.
+ */
+export default function RoomCard () {
+  const router = useRouter()
+  const [filterResults, setFilterResults] = useState(null)
+
+  useEffect(() => {
+    async function load () {
+      try {
+        // get duration to next lectures
+        const startDate = getNextValidDate()
+        const timetable = await getFriendlyTimetable(new Date(), false)
+        const today = timetable.filter(x => x.startDate.getDate() === new Date().getDate())
+
+        if (today.length < 1) {
+          setFilterResults([])
+          return
+        }
+
+        const difference = Math.ceil((today.length === 1 ? (today[0].startDate.getTime() - new Date().getTime()) : (today[1].startDate.getTime() - today[0].endDate.getTime())) / 1000 / 60) - 10 // 10 minutes buffer between lectures
+
+        // no room search if next lecture is less than 15 minutes away (mind the 10 min buffer)
+        if (difference < 5) {
+          setFilterResults([])
+          return
+        }
+
+        // time to next lecture in format hh:mm (minus 10 minutes buffer)
+        const duration = `${Math.floor(difference / 60).toString().padStart(2, '0')}:${(difference % 60).toString().padStart(2, '0')}`
+
+        // filter for suitable rooms
+        const roomRegex = /^[a-zA-Z]+/
+        const building = roomRegex.exec((today.length > 1 ? today[1] : today[0]).raum)[0]
+        let rooms = await filterRooms(formatISODate(startDate), formatISOTime(startDate), BUILDINGS_ALL, duration)
```

Comment:
The `filterRooms` function was previoulsly only used with user input. Here you convert computer given data to weird string formats just to make them fit to the `filterRooms` parameter.

I think splitting the `filterRooms` function into two might be a good idea. One (outer) function which takes user input (i.e. strings) and converts it to data (i.e. Date, minutes as a number etc.) and a second (inner) function which performs the actual filtering would be a good idea.
The existing uses of `filterRooms` can simply use the outer one, while we can use the inner one here.

Reply:
Yeah that makes a lot of sense, will change that :)